velox/functions/sparksql/tests/MakeDecimalTest.cpp

```diff
@@ -0,0 +1,116 @@
+/*
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "velox/common/base/tests/GTestUtils.h"
+#include "velox/functions/sparksql/tests/SparkFunctionBaseTest.h"
+
+namespace facebook::velox::functions::sparksql::test {
+namespace {
+class MakeDecimalTest : public SparkFunctionBaseTest {
+ protected:
+  void testDictionary(
+      const core::CallTypedExprPtr& expr,
+      const VectorPtr& input,
+      const VectorPtr& expected) {
+    // Dictionary vectors as arguments.
+    const auto newSize = input->size() * 2;
+    std::vector<VectorPtr> vectors;
+    vectors.reserve(1);
+    auto indices =
+        makeIndices(newSize, [&](vector_size_t row) { return row / 2; });
+    vectors.emplace_back(
+        VectorTestBase::wrapInDictionary(indices, newSize, input));
+    auto expectedDic =
+        VectorTestBase::wrapInDictionary(indices, newSize, expected);
+    auto actual = evaluate(expr, makeRowVector(vectors));
+    velox::test::assertEqualVectors(expectedDic, actual);
+  }
+
+  void testMakeDecimal(
+      bool nullOnOverflow,
+      const VectorPtr& input,
+      const VectorPtr& expected) {
+    std::vector<core::TypedExprPtr> inputs = {
+        std::make_shared<core::FieldAccessTypedExpr>(input->type(), "c0"),
+        std::make_shared<core::ConstantTypedExpr>(
+            BOOLEAN(), variant(nullOnOverflow))};
+    auto makeDecimal = std::make_shared<const core::CallTypedExpr>(
+        expected->type(), std::move(inputs), "make_decimal");
+    auto result = evaluate(makeDecimal, makeRowVector({input}));
+    velox::test::assertEqualVectors(expected, result);
+    testDictionary(makeDecimal, input, expected);
```

Comment:
Extracted `testConstantEval`(to avoid name duplication with method in derived class) and `testDictionary` to FunctionBaseTest..

Reply:
> Extracted testConstantEval(to avoid name duplication with method in derived class) 

Which method does it conflict with? It would be nice to not have Eval suffix here.

Reply:
Removed this method because `testEncodings` covers constant vector.